src/Shared/QueryStringEnumerable.cs

```diff
@@ -56,6 +57,8 @@ public Enumerator GetEnumerator()
     /// </summary>
     public readonly struct EncodedNameValuePair
     {
+        private static readonly SearchValues<char> DecodingChars = SearchValues.Create("%+");
```

Comment:
Is the use of `SearchValues` beneficial for only two chars (`%` and `+`)?
Isn't `IndexOfAny('%', '+')` faster?

Reply:
In the perf tests I ran for this PR, they were equally fast (results were within the error rate of each other). I repeated the tests multiple times and it looked like a marginal benefit for SearchValues on my CPU, hence I went with that. Do you have other findings or some further details?

Reply:
> they were equally fast (results were within the error rate of each other)

Thanks 👍🏻 

> Do you have other findings or some further details?

I don't have any numbers for this case.
But when looking at the code `MemoryExtensions.IndexOfAny` seems preferable for me, because of
* less methods involved, code goes to vectorized work more directly
* no (static) field needed -- less memory footprint, though this is very little anyway
* as no instance methods are involved I guess it's easier for AOT (especially WASM) to spit out ideal code
* less work for the JIT -- so beneficial in serverless scenarios where AOT isn't used

As said I don't have numbers for these. But for only two chars `SearchValues` seems to heavy for me, I'd just go w/ the plain `IndexOfAny` with the overload taking the two chars.

The motivation for SearchValues were greater sets of needles...although in the meantime that type (and sub-types) got heavily optimized.

Reply:
@MihaZupan can you give an advice whether `SearchValues` could / should be used when there are only two (char) needles used for `IndexOfAny`?

Reply:
SearchValues would call into the same implementation behind the scenes in this case, just with a bit more overhead per call to load the values from fields instead of them being constants.

SearchValues can be faster when you have >= 4 values, I wouldn't bother using it with less as long as the values are constants.

Reply:
Thank you for the suggestion, I will go back to the previous implementation.